### PR TITLE
Fix resolution definition

### PR DESCRIPTION
**Pull request by loicgasser**
FIx for https://github.com/geoadmin/mf-geoadmin3/issues/3514

Bug introduced in
https://github.com/geoadmin/mf-geoadmin3/commit/1820eb9e6febc578ec284e6bf8d2c1d4c70aaccc

I have a hard time believing we had this bug for so long. It's pretty incredible.
I think it explains a lot of bugs we never understood.

So basically we would change the resolution list forever whenever we had a layer with minResolution added. :x:

### CODE REVIEW

**Comment by loicgasser** [comment]
[Test Link](https://mf-geoadmin3.int.bgdi.ch/gal_preview_layers/index.html)

I am even tempted to merge it now lool.

**Comment by loicgasser** [comment]
that possibly explains the terrible perfs we you had all emapis layers as well

File: src/components/map/MapService.js [hunk 1]
```diff
@@ -19,10 +19,16 @@ goog.require('ga_urlutils_service');
 
   module.provider('gaTileGrid', function() {
     var origin = [420000, 350000];
-    var defaultResolutions = [4000, 3750, 3500, 3250, 3000, 2750, 2500, 2250,
-        2000, 1750, 1500, 1250, 1000, 750, 650, 500, 250, 100, 50, 20, 10, 5,
-        2.5, 2, 1.5, 1, 0.5];
-    var wmsResolutions = defaultResolutions.concat([0.25, 0.1]);
```
Comment (oterral) [inline src/components/map/MapService.js]: I'm not sure to undrstand why this doesn't work the `concat` doesn't affect the defaultResolutions array.

Comment (oterral) [inline src/components/map/MapService.js]: ah ok I see the problem is the splice further.

File: src/components/map/MapService.js [hunk 2]
```diff
@@ -43,7 +49,8 @@ goog.require('ga_urlutils_service');
       return {
         get: function(resolutions, minResolution, type) {
           if (!resolutions) {
-            resolutions = (type == 'wms') ? wmsResolutions : defaultResolutions;
+            resolutions = (type == 'wms') ? getWmsResolutions() :
+                getDefaultResolutions();
```
Comment (oterral) [inline src/components/map/MapService.js]: Can you remove the  `resolutions = `    line 58 . It seems useless. Thx.

**Comment by oterral** [comment]
Amazing @loicgasser,  you are the best bug hunter :)  Thx for the fix.

**Comment by cedricmoullet** [comment]
For my understanding: why did it slow the viewer ?

**Comment by oterral** [comment]
At the swiss zoom level, instead of requesting 6 tiles, it requested  hundred of very small tiles at the same time.

File: src/components/map/MapService.js (hunk 2 above)
Comment (oterral) [inline src/components/map/MapService.js]: @loicgasser you're right, it works so leave it as it is.

**Comment by oterral** [comment]
Thx

**Comment by gjn** [comment]
`So basically we would change the resolution list forever whenever we had a layer with minResolution added.`

Could you elaborate a little bit more? I don't think I fully understand.

**Comment by gjn** [comment]
I think this went unnoticed for a long time because we have few layers with minResolution that triggers the slicing.

Good find though.

**Comment by gjn** [comment]
A const language feature would have prevented this. Oh wait, there's const in javascript - in ecmascript 6: 
http://es6-features.org/#Constants

**Pull request merged**